File: Project/Java (Cross Platform)/Power of God/src/power/of/god/Files.java
package power.of.god;


import java.io.File;
import java.util.ArrayList;

/**
 * Created by apotter96 on 4/8/2015.
 */
public class Files {
    private static String mainPath = "power.of.god/";
    private static String[] files = new String[] {
            mainPath,
            mainPath + "userInfo.json", //Old format file
            mainPath + "kjv.xml",
            mainPath + "userInfo.powerinfo" //New user format - still json
    };
    public static String[] filesStr()
    {
        if (!new File(files[0]).exists())
        {
            new File(files[0]).mkdirs(); // Making sure that the directory is ready for use
        }
        return files;
    }
    public static File[] filesObj()
    {
        File[] arrayHere = new File[filesStr().length];
        int loopCount = 0;
        for (String s:filesStr())
        {
            arrayHere[loopCount] = new File(s);
            loopCount++;
        }
        return arrayHere;
    }

}

File: Project/Windows Store/Power of God/PowerOfGodUniversal/Files.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Storage;

namespace PowerOfGodUniversal
{
    public class Files
    {
        private const string _mainPath = "ms-appdata:///local/power.of.god/";

        public static readonly List<string> ListFiles = new List<string>
        {
            _mainPath, // 0
            _mainPath + "userInfo.powerinfo", // 1
            _mainPath + "logins.txt", // 2
            _mainPath + "lastopen.txt", // 3
            _mainPath + "settings.json", // 4
            _mainPath + "KJV.xml", // 5
            _mainPath + "ESV.xml", // 6
            _mainPath + "NIV.xml", // 7
            _mainPath + "NLT.xml", // 8
            _mainPath + "dailyscripture.json", // 9
            _mainPath + "lastds.txt", // 10
            "test.txt" // 11
        };

        public static List<string> FilesStr()
        {
            //TODO CHECK FOR FILE EXISTING
            if (!Directory.Exists(ListFiles.ElementAt(0)))
            {
                Directory.CreateDirectory(ListFiles.ElementAt(0));
            }
            return ListFiles;
        }

        public async static Task<bool> FileExists(int file)
        {
            var folder = StorageFolder.GetFolderFromPathAsync(_mainPath).GetResults();
            try
            {
                var item = await folder.TryGetItemAsync(ListFiles.ElementAt(11));
                return (item != null);
            }
            catch (Exception)
            {
                return false;
            }
        }
        /*public async Task<bool> IfStorageItemExist(StorageFolder folder, string itemName)
        {
            folder = StorageFolder.GetFolderFromPathAsync()
            try
            {
                var item = await folder.TryGetItemAsync(itemName);
                return (item != null);
            }
            catch (Exception ex)
            {
                // Should never get here 
                return false;
            }
        }*/
    }
}
